neuraxle/base.py

```diff
@@ -29,14 +29,55 @@
 from neuraxle.hyperparams.space import HyperparameterSpace, HyperparameterSamples
 
 
-class BaseStep(ABC):
+class Hasher(ABC):
+    def hash(self, hyperparameters: HyperparameterSamples, data_inputs: Any):
+        return hash(data_inputs)
+
+    @abstractmethod
+    def rehash(self, ids, hyperparameters: HyperparameterSamples, data_inputs: Any):
+        raise NotImplementedError()
+
+
+class HasherByIndex(Hasher):
+    def hash(self, hyperparameters, data_inputs: Any):
+        return range(len(data_inputs))
+
+    def rehash(self, ids, hyperparameters: HyperparameterSamples, data_inputs: Any):
+        return self.hash(hyperparameters, data_inputs)
+
+
+class IDHasher(Hasher):
+    def rehash(self, ids, hyperparameters: HyperparameterSamples, data_inputs: Any):
+        return ids
```

Comment:
just delete this one, I'd keep the HasherByID for the pipeline.transform initialization of original ID, and everywhere else I'd use the hasher by hyperparam.